### DISCUSSION TITLE
Feature request: If conflict in file upload exists, show the original

### DISCUSSION THREAD

**Discussion by dougblackjr** [ExpressionEngine/ExpressionEngine#3810] [Feature Requests]
### Suggest an idea for ExpressionEngine

Right now, when a image/file is uploaded EE offers to resolve the conflict by renaming it, appending a numeral, or overwriting. It should also offer the ability to use the original. Would give users the ability to know they've already uploaded the image/file already without reuploading.

- [X] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@dougblackjr I think that option already exists. You can:
a) skip resolving conflict at all
![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/752126/f9f44be6-e7a2-4157-a686-7e81357fd08a)
b) choose "cancel upload" in the popup

Or do you think we need a link to view the file so that people could identify if it's really the same?

**Reply by dougblackjr** [reply]
I think the UI is unclear. So, looking at this, what I'd need to do (if I'm reading this correctly) is:
1. Skip resolve conflict.
2. Go back and choose to use an existing image
3. Search for the existing image and select it

What I think would be nice would be
1. Hit resolve conflict
2. Select Use existing image with same name (or similar nomenclature)

Just to give non-technical users the ability to say "Oh, that's right, I uploaded that, let's use that instead"

**Reply by dougblackjr** [reply]
Great example, I just ran into here. I uploaded this image, but the file manager didn't update. So I saved and re-uploaded and got this:
![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/6020323/00af5036-a910-4256-967f-49cb615afe1e)

**Reply by intoeetive** [reply]
Oh, so this is more relevant for file field, not file manager in general. Got it.

So you're thinking about something like "use existing file" button, correct?

**Reply by dougblackjr** [reply]
Correct on both accounts. I think having an option to use existing file or a button for it would make sense.

**Comment by intoeetive** [comment]
Investigated this a bit, it does not seem to be easy to get the ID / data of the conflicting file, as it's not available in response and also the conflicting uploaded file gets renamed very early in the process. Not saying it's not possible at all, but might be quite tricky.

I wonder if adding "Cancel upload" button directly here would be a good step towards better usability though?

![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/752126/f908fd78-c5eb-425f-b915-b5a93909bd3f)
